All_home_works/hw2/task4.py

```diff
@@ -0,0 +1,35 @@
+"""
+Write a function that accepts another function as an argument. Then it
+should return such a function, so the every call to initial one
+should be cached.
+
+
+def func(a, b):
+    return (a ** b) ** 2
+
+
+cache_func = cache(func)
+
+some = 100, 200
+
+val_1 = cache_func(*some)
+val_2 = cache_func(*some)
+
+assert val_1 is val_2
+
+"""
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    memo = {}
+
+    def wrapper(*args, **kwargs):
+        call = str(args) + str(kwargs)
+        if call in memo:
```

Comment:
You can reorder your code to avoid diplications.

Reply:
You are talking about: "call = of'{args}{kwargs}'" str duplications?
Also, I fixed duplications in "return memo[call]"